app/src/common/shared/org/mozilla/vrbrowser/utils/UrlUtils.java

```diff
@@ -0,0 +1,160 @@
+        }
+
+        return uri.toString();
+    }
+
+    /**
+     * Is the given string a URL or should we perform a search?
+     *
+     * TODO: This is a super simple and probably stupid implementation.
+     */
+    public static boolean isUrl(String url) {
+        String trimmedUrl = url.trim();
+        if (trimmedUrl.contains(" ")) {
+            return false;
+        }
+
+        return trimmedUrl.contains(".") || trimmedUrl.contains(":");
+    }
+
+    public static boolean isValidSearchQueryUrl(String url) {
+        String trimmedUrl = url.trim();
+        if (!trimmedUrl.matches("^.+?://.+?")) {
+            // UI hint url doesn't have http scheme, so add it if necessary
+            trimmedUrl = "http://" + trimmedUrl;
+        }
+
+        final boolean isNetworkUrl = URLUtil.isNetworkUrl(trimmedUrl);
+        final boolean containsToken = trimmedUrl.contains("%s");
+
+        return isNetworkUrl && containsToken;
+    }
+
+    public static boolean isHttpOrHttps(String url) {
+        if (TextUtils.isEmpty(url)) {
+            return false;
+        }
+
+        return url.startsWith("http:") || url.startsWith("https:");
+    }
+
+    public static String stripUserInfo(@Nullable String url) {
+        if (TextUtils.isEmpty(url)) {
+            return "";
+        }
+
+        try {
+            URI uri = new URI(url);
+
+            final String userInfo = uri.getUserInfo();
+            if (userInfo == null) {
+                return url;
+            }
+
+            // Strip the userInfo to minimise spoofing ability. This only affects what's shown
+            // during browsing, this information isn't used when we start editing the URL:
+            uri = new URI(uri.getScheme(), null, uri.getHost(), uri.getPort(), uri.getPath(), uri.getQuery(), uri.getFragment());
+
+            return uri.toString();
+        } catch (URISyntaxException e) {
+            // We might be trying to display a user-entered URL (which could plausibly contain errors),
+            // in this case its safe to just return the raw input.
+            // There are also some special cases that URI can't handle, such as "http:" by itself.
+            return url;
+        }
+    }
+
+    public static boolean isPermittedResourceProtocol(@Nullable final String scheme) {
+        return scheme != null && (
+                scheme.startsWith("http") ||
+                        scheme.startsWith("https") ||
+                        scheme.startsWith("file") ||
+                        scheme.startsWith("data") ||
+                        scheme.startsWith("javascript") ||
+                        scheme.startsWith("about"));
+    }
+
+    public static boolean isSupportedProtocol(@Nullable final String scheme) {
+        return scheme != null && (isPermittedResourceProtocol(scheme) || scheme.startsWith("error"));
+    }
+
+    public static boolean isInternalErrorURL(final String url) {
+        return "data:text/html;charset=utf-8;base64,".equals(url);
```

Comment:
we need to figure out a better way of identifying an error URL. there can be data URIs that look like this that aren't error URIs.